nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/users/SuggestUserEntries.java

```diff
@@ -269,8 +276,20 @@ public Blob run() throws IOException {
                 user = userManager.getUserModel(userId);
                 UserAdapter userAdapter = user.getAdapter(UserAdapter.class);
                 List<String> groups = userAdapter.getGroups();
-                if (groups != null && groups.contains(groupRestriction)) {
-                    result.add(obj);
+                List<String> restrictedGroups = new ArrayList<>();
+                restrictedGroups.add(groupRestriction);
+
+                if (CollectionUtils.isNotEmpty(groups)) {
+                    if (allowSubGroupsRestriction) {
+                        List<String> subGroups = userManager.getDescendantGroups(groupRestriction);
+                        for (String subGroup : subGroups) {
+                            restrictedGroups.add(subGroup);
+                        }
+                    }
+                    groups.retainAll(restrictedGroups);
+                    if (!groups.isEmpty()) {
```

Comment:
I'd rather not modify `groups` as it comes from the adapter and I'm not sure we have enough defensive copies (and actually there's nothing that guarantees that's it's a modifiable list).
As the check we want is that the intersection is not empty, I think this works just as well:
```
restrictedGroups.retainAll(groups);
if (!restrictedGroups.isEmpty()) {
```
But of course run the unit tests to make sure.